### PR TITLE
Diffui show heapinfo

### PR DESCRIPTION
**Pull request by davesu**

### CODE REVIEW

File: hollow-diff-ui/src/main/java/com/netflix/hollow/HollowDiffUtil.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.netflix.hollow;
+
+import java.text.DecimalFormat;
+
+public class HollowDiffUtil {
+    private static final String[] HEAP_SIZE_UNITS = new String[] { "B", "KB", "MB", "GB", "TB" };
+    private static final DecimalFormat BYTE_FORMATTER = new DecimalFormat("#,##0.#");
+
+    public static String formatBytes(long sizeInBytes) {
```
Comment (akhaku) [inline hollow-diff-ui/src/main/java/com/netflix/hollow/HollowDiffUtil.java]: would be nice to have a couple unit tests for this method :)

File: hollow-diff-ui/src/main/java/com/netflix/hollow/HollowDiffUtil.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        if (sizeInBytes==0) return "0";
+
+        String sign = (sizeInBytes < 0) ? "-" : "";
+        sizeInBytes = Math.abs(sizeInBytes);
+        int digitGroups = (int) (Math.log10(sizeInBytes)/Math.log10(1024));
```
Comment (akhaku) [inline hollow-diff-ui/src/main/java/com/netflix/hollow/HollowDiffUtil.java]: _technically_ this is KiB/MiB etc and not KB/MB (as called out in the comments at https://stackoverflow.com/questions/3263892/format-file-size-as-mb-gb-etc)

File: hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/pages/DiffPage.java [hunk 3]
```diff
@@ -63,6 +65,15 @@ public void render(HttpServletRequest req, HollowUISession session, Writer write
...
         ctx.put("toBlobName", diffUI.getToBlobName());
 
+        HollowDiff diff = diffUI.getDiff();
+        long heapFrom = diff.getFromStateEngine().calcApproxDataSize();
+        long heapTo = diff.getToStateEngine().calcApproxDataSize();
+        long heapDiff = heapTo-heapFrom;
+        ctx.put("fromHeap", formatBytes(heapFrom));
+        ctx.put("toHeap", formatBytes(heapTo));
+        ctx.put("diffHeap", (heapDiff > 0 ? "+" : "") + formatBytes(heapDiff));
+        ctx.put("diffHeap_cssClass", heapDiff<=0 ? "heap_dec" : "heap_inc");
```
Comment (akhaku) [inline hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/pages/DiffPage.java:75]: nice

**Review by akhaku — APPROVED** [review]
LGTM with the comments, I'll let Sunjeet do a once-over too though

File: hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/pages/DiffOverviewPage.java [hunk 4]
```diff
@@ -107,6 +113,34 @@ public int compare(HollowDiffOverviewTypeEntry o1, HollowDiffOverviewTypeEntry o
                     return o2.getTotalInTo() - o1.getTotalInTo();
                 }
             });
+        } else if("fromHeap".equals(sortBy)) {
+            Collections.sort(overviewEntries, new Comparator<HollowDiffOverviewTypeEntry>() {
```
Comment (Sunjeet) [inline hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/pages/DiffOverviewPage.java]: nit: java8-ify with
```
Collections.sort(overviewEntries, (o1, o2) -> (int)(o2.getHeapInFrom() - o1.getHeapInFrom()));
```

Comment (akhaku) [reply hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/pages/DiffOverviewPage.java]: could also replace with Comparators.comparingInt, eg
```
Collections.sort(overviewEntries, Comparator.comparingInt(HollowDiffOverviewTypeEntry::getHeapInFrom))
```

**Review by akhaku — APPROVED** [review]

**Pull request merged**